asks/request_object.py

```diff
@@ -244,6 +244,11 @@ async def make_request(self, redirect=False):
         if self.streaming:
             return None, response_obj
 
+        if 'connection' in asks_headers and asks_headers['connection'] == 'close' and self.sock._active:
+            self.sock._active = False
```

Comment:
"Deactivating" the socket does not let it back into the pool, but it would not be able to get there anyway as it is not returned to the caller. We could either mark the socket as inactive and return it or do not mark it and return `None`.